prebuilt/meson.build

```diff
@@ -14,12 +14,117 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
+# We do require very specific layout for 'prebuilt-path':
+#
+# * For Linux:
+#     get_option('prebuilt-path') / opt / bin     : executables
+#     get_option('prebuilt-path') / opt / include : include files
+#     get_option('prebuilt-path') / opt / lib     : dynamic and static libraries
+#   NOTE: basically 'prebuilt-path' is DESTDIR. You can install it with:
+#     COPY /opt/dist /
+#
+# * For Windows:
+#     get_option('prebuilt-path') / bin           : executables
+#     get_option('prebuilt-path') / include       : include files
+#     get_option('prebuilt-path') / lib           : dynamic and static libraries
+
+prebuilt_path = get_option('prebuilt-path')
+
+# find_library() requires absolute paths. include_directories() requires
+# relative paths for project source directories and accepts absolute paths (as
+# it seems) for external directories.
+if prebuilt_path != ''
+  if fs.is_absolute(prebuilt_path)
+    prebuilt_path_abs = prebuilt_path
+  else
+    prebuilt_path_abs = meson.source_root() / prebuilt_path
+    # recalculate relative path from the current directory
+    prebuilt_path = '..' / prebuilt_path
+  endif
+else
+  prebuilt_path = host_machine.system() + '-' + build_machine.cpu_family()
+  prebuilt_path_abs = meson.current_source_dir() / prebuilt_path
+endif
+
+if not fs.exists(prebuilt_path)
+  warning('Invalid prebuilt-path provided.',
+    'Project might fail to build due to missing dependencies.')
+  subdir_done()
```

Comment:
The original closed sourced commit that https://github.com/intel/cloud-streaming/pull/55/commits/13a987b5b9736c94b3299bf294a2aff2e35d5035 originates from had a fairly verbose comment here explaining the logic. Assuming this was intentionally not added?

Reply:
Yes, this was dropped intentionally. In internal repository we have few lines here which add subproject used to fetch prebuilt-binary dependencies. We can't use same subproject here since it fetches from internal locations not visible outside. I am still thinking whether we can have subproject line here in open source - that's one place I will be considering further for closer internal/public alignment.